engine/runtime/src/main/java/org/enso/interpreter/runtime/number/EnsoBigInteger.java

```diff
@@ -44,6 +46,71 @@ String toDisplayString(boolean allowSideEffects) {
     return value.toString();
   }
 
+  @ExportMessage
+  boolean isNumber() {
+    return true;
+  }
+
+  @ExportMessage
+  final boolean fitsInByte() {
+    return false;
+  }
+
+  @ExportMessage
+  final boolean fitsInShort() {
+    return false;
+  }
+
+  @ExportMessage
+  final boolean fitsInInt() {
+    return false;
+  }
+
+  @ExportMessage
+  final boolean fitsInLong() {
+    return false;
+  }
+
+  @ExportMessage
+  final boolean fitsInFloat() {
+    return false;
+  }
+
+  @ExportMessage
+  final boolean fitsInDouble() {
+    return true;
+  }
+
+  @ExportMessage
+  final byte asByte() throws UnsupportedMessageException {
+    return (byte) 0;
+  }
+
+  @ExportMessage
+  final short asShort() throws UnsupportedMessageException {
+    return (short) 0;
+  }
+
+  @ExportMessage
+  final int asInt() throws UnsupportedMessageException {
+    return 0;
+  }
+
+  @ExportMessage
+  final long asLong() throws UnsupportedMessageException {
+    return 0L;
+  }
+
+  @ExportMessage
+  final float asFloat() throws UnsupportedMessageException {
+    return 0.0F;
+  }
+
+  @ExportMessage
+  final double asDouble() throws UnsupportedMessageException {
+    return BigIntegerOps.toDouble(value);
+  }
```

Comment:
I think that these should throw `UnsupportedMessageException`, as the documentation says that `Returns the receiver value as Java X primitive if the number fits without loss of precision` - but we explicitly say that `fitsInX() == false` and indeed our bigints do not fit into these.